hbase-server/src/main/java/org/apache/hadoop/hbase/util/FSUtils.java

```diff
@@ -965,11 +964,10 @@ public abstract void recoverFileLease(final FileSystem fs, final Path p,
 
   public static List<Path> getTableDirs(final FileSystem fs, final Path rootdir)
       throws IOException {
-    List<Path> tableDirs = new LinkedList<>();
+    List<Path> tableDirs = new ArrayList<>();
```

Comment:
what's your reasoning that ArrayList is a better choice here?

Reply:
Generally speaking ArrayList should be the default choice.

https://stackoverflow.com/questions/322715/when-to-use-linkedlist-over-arraylist-in-java

In this case, it makes sense because it there is the advantage of using `addAll` method on the Collection.  The array can be resized very accurately since it's not coming in one-by-one.  Also, if the list returned is large, it is much faster to copy the references into `tableDirs` instead of having to instantiate a LinkedList Node for each value.

Reply:
that sounds reasonable. we're more likely to have a small number of namespaces with lots of tables then have lots of namespaces with very few tables.